### EDA
Let's explore the data a bit! Before continuing...
Extract the 'Close' prices from the 'Stock Info' level of the DataFrame and then calculate the maximum value for each bank's stock. 

In [27]:
# Import relevant libraries
import pandas as pd
import numpy as np
import seaborn as sns
import datetime as dt
%matplotlib inline

In [24]:
pwd

'/home/jupyter-raphrivers/Historical-Stock-Data-Analysis'

In [34]:
# import cleaned dataframe
bstocks = ('/home/jupyter-raphrivers/Historical-Stock-Data-Analysis/assets/bankstocks.csv')
bankStocks = pd.read_csv(bstocks, header=[0, 1])

In [38]:
# Set the column name levels
bankStocks.columns.names = ['Ticker','Stock Info']

In [35]:
bankStocks.head()

Ticker        BAC                                              \
   Stock Info       Open       High        Low      Close  Adj Close   
0        Date        NaN        NaN        NaN        NaN        NaN   
1  2018-01-02  29.750000  29.900000  29.610001  29.900000  25.995729   
2  2018-01-03  29.900000  29.940001  29.690001  29.799999  25.908791   
3  2018-01-04  29.969999  30.440001  29.879999  30.190001  26.247868   
4  2018-01-05  30.370001  30.420000  30.049999  30.330000  26.369587   

                       C                        ...         MS             \
       Volume       Open       High        Low  ...        Low      Close   
0         NaN        NaN        NaN        NaN  ...        NaN        NaN   
1  57121600.0  75.089996  75.180000  74.019997  ...  52.169998  52.189999   
2  57865700.0  74.349998  74.739998  73.970001  ...  51.919998  52.330002   
3  76512500.0  75.010002  76.110001  74.660004  ...  52.509998  53.130001   
4  56445200.0  75.709999  75.769997  74.959999  ...  52.740002  53.110001   

                               WFC                                   \
   Adj Close     Volume       Open       High        Low      Close   
0        NaN        NaN        NaN        NaN        NaN        NaN   
1  43.752384  6644600.0  61.040001  61.360001  60.700001  61.090000   
2  43.869759  6955500.0  61.220001  61.770000  61.099998  61.560001   
3  44.540417  8678800.0  61.980000  63.049999  61.910000  62.330002   
4  44.523651  6476000.0  62.759998  62.810001  62.090000  62.750000   

                          
   Adj Close      Volume  
0        NaN         NaN  
1  51.275253  13819300.0  
2  51.669731  14203700.0  
3  52.316029  18740500.0  
4  52.668556  14217900.0  

[5 rows x 37 columns]

In [39]:
# Maximum Closing price for each bank's stock throughout the time period
bankStocks.xs(key='Close', axis=1, level='Stock Info').max()

Ticker
BAC     49.380001
C       81.910004
GS     423.850006
JPM    171.779999
MS     108.730003
WFC     65.930000
dtype: float64

The highest closing price was Goldman Sach (GS) closing at USD 423.85 dollars, while the next highest is JP Morgan (JPM) closing at USD 171.78

In [40]:
# Minimum Closing price for each bank's stock throughout the time period
bankStocks.xs(key='Close', axis=1, level='Stock Info').min()

Ticker
BAC     18.080000
C       35.389999
GS     134.970001
JPM     79.029999
MS      27.809999
WFC     21.139999
dtype: float64

The lowest closing price for the period was Bank of America (BAC) with stock closing at USD 18.08, while the lowest for Citigroup (C) was USD 35.39, 